pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/avatar_utils.rs

```diff
@@ -1124,6 +1155,37 @@ impl AvatarUtils {
 
 		result
 	}
+
+	pub fn indexes_of_max(byte_array: &[u8]) -> Vec<usize> {
+		let mut max_value = u8::MIN;
+		let mut max_indexes = Vec::new();
+
+		for (i, byte) in byte_array.iter().enumerate() {
+			match byte.cmp(&max_value) {
+				Ordering::Greater => {
+					max_value = *byte;
+					max_indexes = vec![i];
```

Comment:
this is not usable ... will throw an error

max_indexes = Vec::new();  // Initialize a new Vec
max_indexes.push(i);  // Push i into the new Vec

Reply:
Can we refactor this algorithm so that it
1. finds the maximum value first
2. find the indexes of that value

This will involve iterating twice but we don't need to worry about performance tradeoffs since the `byte_array` is not too big.

Reply:
I replaced it with `Vec::clear()` which has negligible performance impact.